### PR TITLE
Favorite episodes/favorite characters page

### PR DESCRIPTION
**Pull request by Efekahya**

### CODE REVIEW

File: src/context/favoriteContext.tsx [hunk 1]
```diff
@@ -1,6 +1,5 @@
-import React, { useEffect } from "react";
+import React from "react";
 import { IFavoriteContext } from "../types/contextInterface";
```
Comment (atvdi) [inline src/context/favoriteContext.tsx:2]: You are declaring that this is already type definition folder in the folder name.
```suggestion
import { IFavoriteContext } from "../types/context";
```

File: src/pages/CharacterDetails/styles.scss [hunk 2]
```diff
@@ -210,11 +210,15 @@
...
+        max-height: 132px;
         overflow: hidden;
+        width: calc(100% - 64px);
+        padding: 32px;
+        margin: -32px;
     }
 
     .character-episode-item {
-        width: calc(50% - 16px);
+        width: calc(50% - 12px);
```
Comment (atvdi) [inline src/pages/CharacterDetails/styles.scss:221]: You don't have to but I would use `gap` in the parent for this one.

File: src/pages/CharacterDetails/styles.scss [hunk 3]
```diff
@@ -210,11 +210,15 @@
 
     .character-episode-items {
         flex-wrap: wrap;
+        max-height: 132px;
         overflow: hidden;
+        width: calc(100% - 64px);
+        padding: 32px;
+        margin: -32px;
```
Comment (atvdi) [inline src/pages/CharacterDetails/styles.scss:217]: What is the purpose of these three lines? You can just give horizontal margin.

File: src/pages/Episodes/styles.scss [hunk 4]
```diff
@@ -63,6 +63,15 @@
     max-width: 1120px;
 }
 
+
+.episodes-page-loading-fixed {
```
Comment (atvdi) [inline src/pages/Episodes/styles.scss:67]: Why did you use the term `fixed` here?

File: src/pages/Favorites/index.tsx [hunk 5]
```diff
@@ -62,20 +66,31 @@ export default function Favorites() {
       });
     });
   }, [episodes, favoritedItems]);
+  useEffect(() => {
+    setFilteredCharacters(() => {
+      return characters.filter(charater => {
```
Comment (atvdi) [inline src/pages/Favorites/index.tsx:71]:
```suggestion
      return characters.filter(character => {
```

File: src/pages/Episodes/styles.scss (hunk 4 above)
Comment (Efekahya) [reply src/pages/Episodes/styles.scss:67]: position:fixed;

File: src/pages/CharacterDetails/styles.scss (hunk 3 above)
Comment (Efekahya) [reply src/pages/CharacterDetails/styles.scss:217]: padding is given for box shadow to not clip by its parent and margin is given to fix its position I couldn't find any clever solutions for this to work

**Pull request merged**